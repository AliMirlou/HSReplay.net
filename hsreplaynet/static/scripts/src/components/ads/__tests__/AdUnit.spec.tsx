import AdUnit from "../AdUnit";
import renderer from "react-test-renderer";
import * as React from "react";
import UserData from "../../../UserData";

describe("AdUnit", () => {
	test("does render when ad feature is enabled", () => {
		UserData.hasFeature = feature => feature === "ads";
		const component = renderer.create(<AdUnit id="ad-42" size="300x250" />);
		const tree = component.toJSON();
		expect(tree).toBeDefined();
		expect(tree).toMatchSnapshot();
	});

	test("does not render when ad feature is disabled", () => {
		UserData.hasFeature = () => false;
		const component = renderer.create(<AdUnit id="ad-42" size="300x250" />);
		expect(component.toJSON()).toBeNull();
	});

	test("does not render for premium user", () => {
		UserData.hasFeature = feature => feature === "ads";
		UserData.isPremium = () => true;
		const component = renderer.create(<AdUnit id="ad-42" size="300x250" />);
		expect(component.toJSON()).toBeNull();
	});

	test("does render correctly in debug mode", () => {
		UserData.hasFeature = feature =>
			["ads", "ads-debug"].indexOf(feature) !== -1;
		const component = renderer.create(<AdUnit id="ad-42" size="300x250" />);
		const tree = component.toJSON();
		expect(tree.children[0].children[0]).toBe("ad-42");
		expect(tree).toBeDefined();
		expect(tree).toMatchSnapshot();
	});
});
