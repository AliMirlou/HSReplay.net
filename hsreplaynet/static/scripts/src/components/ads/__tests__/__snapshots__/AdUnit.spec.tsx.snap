// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdUnit does render correctly in debug mode 1`] = `
<div
  className="ad-unit ad-unit--desktop"
  id="ad-42"
  style={
    Object {
      "height": "250px",
      "width": "300px",
    }
  }
>
  <p
    style={
      Object {
        "fontWeight": "bold",
        "lineHeight": "250px",
        "textAlign": "center",
      }
    }
  >
    ad-42
  </p>
</div>
`;

exports[`AdUnit does render when ad feature is enabled 1`] = `
<div
  className="ad-unit ad-unit--desktop"
  id="ad-42"
  style={
    Object {
      "height": "250px",
      "width": "300px",
    }
  }
/>
`;
